sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -373,7 +373,7 @@ case class InsertIntoTable(
       // Note: The parser (visitPartitionSpec in AstBuilder) already turns
       // keys in partition to their lowercase forms.
       val staticPartCols = partition.filter(_._2.isDefined).keySet
-      Some(table.output.filterNot(a => staticPartCols.contains(a.name)))
+      Some(table.output.filterNot(a => staticPartCols.contains(a.name.toLowerCase)))
```

Comment:
If `visitPartitionSpec` change the keys in partition spec to lower case, it sounds we are having multiple bugs in name comparison for partition columns.